tests/git_edit_index_tests.py

```diff
@@ -396,6 +423,23 @@ def test_performs_correct_action_when_untracked_file_is_to_be_deleted(self):
         self.remove.assert_called_once_with('file.txt')
         self.assertFalse(self.perform_git_action.called)
 
+    def test_performs_correct_action_when_ignored_file_is_to_be_added(self):
+        orig_entry = IndexEntry('!', 'file.txt')
+        new_entry = IndexEntry('A', 'file.txt')
+
+        reflect_index_change(orig_entry, new_entry)
+
+        self.perform_git_action.assert_called_once_with('add', 'file.txt')
```

Comment:
I think the `git` call here should be `git add -f file.txt`. Otherwise, a bare `git add` without `-f` will complain:
```
The following paths are ignored by one of your .gitignore files:
file.txt
Use -f if you really want to add them.
```